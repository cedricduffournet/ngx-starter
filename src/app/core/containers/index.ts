export * from './toaster';
export * from './http-progress-bar/http-progress-bar.component';
