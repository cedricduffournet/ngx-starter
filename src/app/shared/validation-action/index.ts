export * from './validation-action.module';
