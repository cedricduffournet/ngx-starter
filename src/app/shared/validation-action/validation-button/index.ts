export * from './validation-button.component';
