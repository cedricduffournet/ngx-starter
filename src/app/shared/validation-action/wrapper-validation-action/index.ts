export * from './wrapper-validation-action.component';
