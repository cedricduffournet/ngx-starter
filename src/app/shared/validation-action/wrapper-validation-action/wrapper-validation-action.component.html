<div class="d-flex justify-content-end align-items-center p-3">
  <ng-content></ng-content>
</div>
