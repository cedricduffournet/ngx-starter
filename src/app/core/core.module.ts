import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { environment } from '@env/environment';
import {
  HttpClientModule,
  HttpClient,
  HTTP_INTERCEPTORS
} from '@angular/common/http';

import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';
import { ModalModule } from 'ngx-bootstrap/modal';
import { ToasterModule } from 'angular2-toaster';

import {
  ToasterComponent,
  HttpProgressBarComponent
} from '@app/core/containers';
import { HttpProgressInterceptor } from '@app/core/services/http-progress.interceptor';

const MODULES: any[] = [
  HttpClientModule,
  TranslateModule.forRoot({
    loader: {
      provide: TranslateLoader,
      useFactory: HttpLoaderFactory,
      deps: [HttpClient]
    }
  }),
  ModalModule.forRoot(),
  ToasterModule.forRoot(),
  CommonModule
];

export const COMPONENTS = [ToasterComponent, HttpProgressBarComponent];

@NgModule({
  imports: MODULES,
  declarations: COMPONENTS,
  exports: COMPONENTS,
  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpProgressInterceptor,
      multi: true
    }
  ]
})
export class CoreModule {
  constructor(
    @Optional()
    @SkipSelf()
    parentModule: CoreModule
  ) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import only once in main AppModule.'
      );
    }
  }
}

export function HttpLoaderFactory(http: HttpClient) {
  return new TranslateHttpLoader(
    http,
    `${environment.prefixAsset}/assets/i18n/`,
    '.json'
  );
}
