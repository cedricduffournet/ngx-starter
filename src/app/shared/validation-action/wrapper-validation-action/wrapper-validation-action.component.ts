import { Component, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'app-wrapper-validation-action',
  templateUrl: './wrapper-validation-action.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class WrapperValidationActionComponent {}
