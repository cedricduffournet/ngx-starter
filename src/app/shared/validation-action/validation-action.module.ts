import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TranslateModule } from '@ngx-translate/core';

import { ValidationButtonComponent } from '@app/shared/validation-action/validation-button';
import { WrapperValidationActionComponent } from '@app/shared/validation-action/wrapper-validation-action';

@NgModule({
  imports: [CommonModule, TranslateModule],
  exports: [ValidationButtonComponent, WrapperValidationActionComponent],
  declarations: [ValidationButtonComponent, WrapperValidationActionComponent]
})
export class ValidationActionModule {}
