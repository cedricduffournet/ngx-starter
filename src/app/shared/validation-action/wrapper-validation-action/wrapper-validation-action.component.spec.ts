import { TestBed, ComponentFixture } from '@angular/core/testing';

import { WrapperValidationActionComponent } from '@app/shared/validation-action/wrapper-validation-action';

describe('WrapperValidationActionComponent', () => {
  let fixture: ComponentFixture<WrapperValidationActionComponent>;
  let component: WrapperValidationActionComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [WrapperValidationActionComponent]
    });
    fixture = TestBed.createComponent(WrapperValidationActionComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
