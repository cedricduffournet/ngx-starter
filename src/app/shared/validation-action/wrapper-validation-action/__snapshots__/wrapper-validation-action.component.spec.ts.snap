// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WrapperValidationActionComponent should be created 1`] = `
<app-wrapper-validation-action>
  <div
    class="d-flex justify-content-end align-items-center p-3"
  />
</app-wrapper-validation-action>
`;
